text/0001-agile-coretime.md

```diff
@@ -0,0 +1,652 @@
+| `INTERLUDE_PERIOD`  | `7 * DAYS`                   | specified  |
+| `LEADIN_PERIOD`     | `7 * DAYS`                   | specified  |
+| `TIMESLICE`         | `8 * MINUTES`                | specified  |
+| `BULK_TARGET`       | `30`                         | suggested  |
+| `BULK_LIMIT`        | `45`                         | suggested  |
+| `RENEWAL_PRICE_CAP` | `Perbill::from_percent(2)`   | suggested  |
+
+
+#### Instantaneous Price Progression
+
+This proposal assumes the existence of a Relay-chain-based price-setting mechanism for the Instantaneous Coretime Market which alters from block to block, taking into account several parameters: the last price, the size of the Instantaneous Coretime Pool (in terms of cores per Relay-chain block) and the amount of Instantaneous Coretime waiting for processing (in terms of Core-blocks queued).
+
+The ideal situation is to have the size of the Instantaneous Coretime Pool be equal to some factor of the Instantaneous Coretime waiting. This allows all Instantaneous Coretime sales to be processed with some limited latency while giving limited flexibility over ordering to the Relay-chain apparatus which is needed for efficient operation.
+
+If we set a factor of three, and thus aim to retain a queue of Instantaneous Coretime Sales which can be processed within three Relay-chain blocks, then we would increase the price if the queue goes above three times the amount of cores available, and decrease if it goes under.
+
+Let us assume the values `OLD_PRICE`, `FACTOR`, `QUEUE_SIZE` and `POOL_SIZE`. A simple definition of the `NEW_PRICE` would be thus:
+
+```
+NEW_PRICE := IF QUEUE_SIZE < POOL_SIZE * FACTOR THEN
+    OLD_PRICE * 0.95
+ELSE
+    OLD_PRICE / 0.95
+END IF
+```
+
+This exists only as a trivial example to demonstrate a basic solution exists, and should not be intended as a concrete proposal.
+
+### Notes on Types
+
+This exists only as a short illustration of a potential technical implementation and should not be treated as anything more.
+
+#### Regions
+
+This data schema achieves a number of goals:
+- Coretime can be individually traded at a level of a single usage of a single core.
+- Coretime Regions, of arbitrary span and up to 1/80th interlacing can be exposed as NFTs and exchanged.
+- Any Coretime Region can be contributed to the Instantaneous Coretime Pool.
+- Unlimited number of individual Coretime contributors to the Instantaneous Coretime Pool. (Effectively limited only in number of cores and interlacing level; with current values this would allow 80,000 individual payees per timeslice).
+- All keys are self-describing.
+- Workload to communicate core (re-)assignments is well-bounded and low in weight.
+- All mandatory bookkeeping workload is well-bounded in weight.
+
+```rust
+type Timeslice = u32; // 80 block amounts.
+type CoreIndex = u16;
+type CoreMask = [u8; 10]; // 80-bit bitmap.
+
+// 128-bit (16 bytes)
+struct RegionId {
```

Comment:
as an implementation note, out-of-scope for the RFC, but these datatypes depend on using a packed representation as opposed to standard alignments

Reply:
I did make a test to check the size (at least on my 64-bit M1 architecture) and it was indeed 128-bit as expected without any explicit packing.